SimpleRss/SimpleRss/Services/XMLParserService.swift

```diff
@@ -0,0 +1,73 @@
+//
+//  RssParser.swift
+//  SimpleRss
+//
+//  Created by Voldem on 11/21/18.
+//  Copyright © 2018 Vladimir Koptev. All rights reserved.
+//
+
+import Foundation
+
+class XMLParserService: NSObject {
+    var parser: XMLParser!
+    var feedList = [Feed]()
+    var feed: Feed?
+    var foundElementName: String?
+    var error: Error? = nil
+    
+    func fetchXMLData(for url: String, withCallback completionHandler: @escaping (_ result: [Feed]?, _ error: Error?) -> Void) {
+        parser = XMLParser(contentsOf: URL(string: url)!)!
+        parser.delegate = self
+        
+        if parser.parse() {
+            if error != nil {
+                completionHandler(nil, error)
+            }
+            else {
+                completionHandler(feedList, nil)
+            }
+        }
+        else {
+            let newError = NSError(domain:"", code: 400, userInfo: [NSLocalizedDescriptionKey: "Error while parsing"])
+            completionHandler(nil, newError)
+        }
+    }
+}
+
+extension XMLParserService: XMLParserDelegate {
+    func parser(_ parser: XMLParser, didStartElement elementName: String, namespaceURI: String?, qualifiedName qName: String?, attributes attributeDict: [String : String]) {
+        
+        foundElementName = elementName
+        if elementName == "item" {
+            feed = Feed()
+            foundElementName = ""
+        }
+        else if foundElementName == "media:content" {
+            feed?.picUrl = attributeDict["url"]!
+        }
+    }
+    
+    func parser(_ parser: XMLParser, foundCharacters string: String) {
+        if foundElementName == "title" {
```

Comment:
`guard let` would be very helpful in such cases, when a method has no sense without a valid variable.